### PR TITLE
Update angular-cookies to 1.8.x

### PR DESCRIPTION
**Pull request by thg-ryan-jackson**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.angularjs.org/api/ngCookies
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## angular-cookies ([<kbd>unpkg</kbd>](https://unpkg.com/browse/angular-cookies@latest/))
was missing the following properties:
1. padStart
1. padEnd
1. trimLeft
1. trimRight
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5d20477103939c93553e731de2a4ade1fa275573
</p>

File: types/angular-cookies/index.d.ts [hunk 1]
```diff
@@ -67,7 +86,7 @@ declare module 'angular' {
 
         /**
          * CookieStoreService DEPRECATED
-         * see https://code.angularjs.org/1.2.26/docs/api/ngCookies/service/$cookieStore
+         * see https://code.angularjs.org/1.3.20/docs/api/ngCookies/service/$cookieStore
```
Comment (peterblazejewicz) [inline types/angular-cookies/index.d.ts:89]: AFAIK the notion of store needs to be completely removed in order to match 1.8:
https://github.com/angular/angular.js/blob/e55d352e942465479fa9f93b566db20a96b4cf15/docs/content/guide/migration.ngdoc#L453
There is only single public api now for cookies:
https://github.com/angular/angular.js/blob/master/src/ngCookies/cookies.js
https://code.angularjs.org/1.8.0/docs/api/ngCookies/service/$cookies

File: types/angular-cookies/index.d.ts [hunk 2]
```diff
@@ -41,19 +52,27 @@ declare module 'angular' {
...
+             * Accepts the values "lax" and "strict".
+             */
+            samesite?: string;
         }
 
         /**
-         * CookieService
-         * see http://docs.angularjs.org/api/ngCookies.$cookies
+         * CookiesService
+         * see https://code.angularjs.org/1.3.20/docs/api/ngCookies/service/$cookies
```
Comment (peterblazejewicz) [inline types/angular-cookies/index.d.ts:64]:
```suggestion
         * see https://code.angularjs.org/1.8.0/docs/api/ngCookies/service/$cookies
```

File: types/angular-cookies/index.d.ts [hunk 3]
```diff
@@ -41,19 +52,27 @@ declare module 'angular' {
...
          */
         interface ICookiesService {
+            /**
+             * DEPRECATED
+             */
             [index: string]: any;
```
Comment (peterblazejewicz) [inline types/angular-cookies/index.d.ts:70]: and this lines should be probably removed as well:
https://github.com/angular/angular.js/blob/e55d352e942465479fa9f93b566db20a96b4cf15/docs/content/guide/migration.ngdoc#L453
only public setters/getters are now possible I thik according to that line:
>Note that
for object values you need to use the `putObject` & `getObject` methods, as
`get`/`put` will not correctly save/retrieve the object values.

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
@thg-ryan-jackson pls review my comments! Thx!

**Comment by thg-ryan-jackson** [comment]
@peterblazejewicz thanks for reviewing! I've pushed a commit address your comments

is it okay that removing those types (eg. CookieStoreService) is a breaking change but only published as a minor version? (angular didn't follow semver until 2.0.0). I initially didn't remove those deprecated types for this reason

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@thg-ryan-jackson  thx for the contribution!

**Comment by thg-ryan-jackson** [comment]
Ready to merge

**Pull request merged**